phonon/model.py

```diff
@@ -0,0 +1,35 @@
+import phonon.fields
+import phonon.exceptions
+
+
+class MetaModel(type):
+    def __new__(cls, name, parents, dct):
+        cls._fields = {}
+        for key, val in dct.items():
+            if isinstance(val, phonon.fields.Field):
+                cls._fields[key] = val
+
+        return super(MetaModel, cls).__new__(cls, name, parents, dct)
+
+
+class Model(object):
```

Comment:
You may have noticed we don't define an `execute` operation. This requires reference counting, which is not yet implemented here. It's a little contrary to the async, lock-free approach to things. Let's chat about how this could work soon.